src/shared/scaling/scaleArray.js

```diff
@@ -1,20 +1,45 @@
-export default function scaleArray(array, scalingParameters) {
+export default function scaleArray(
+  imageFrame,
+  preScaleOptions,
+  array,
+  scalingParameters
+) {
   const arrayLength = array.length;
-  const { rescaleSlope, rescaleIntercept, suvbw } = scalingParameters;
+  const rescaleSlope = scalingParameters?.rescaleSlope || 1;
+  const rescaleIntercept = scalingParameters?.rescaleIntercept || 0;
+  const suvbw = scalingParameters?.suvbw || 1;
```

Comment:
why should we go through headache of the scale and intercept for large arrays if they are undefiend? can we return early instead? Also don't think we get to here at all if `rescaleSlope` and `rescaleIntercept` are not numbers because of the check in `decodeImageFrame`.

Reply:
I see, I guess we can say that in this function always `typeof rescaleSlope === 'number'` is true, so no need for `scalingParameters?.rescaleSlope || 1;` right?
I guess you are right about the PT scenario

Reply:
Figured for readability I just follow the PET case since it doesn’t change the functionality.

Reply:
I’m okay either way, I can remove the || defaults too